apps/vessel-history/features/activity-by-type/activity-by-type.selectors.ts

```diff
@@ -0,0 +1,72 @@
+import { createSelector } from '@reduxjs/toolkit'
+import { DateTime } from 'luxon'
+import { t } from 'features/i18n/i18n'
+import { selectFilteredEvents } from 'features/vessels/activity/vessels-activity.selectors'
+import { PortVisitSubEvent } from 'types/activity'
+import { EventTypes } from '../../../../libs/api-types/src/events'
+
+export const selectFilteredEventsWithMainPortVisit = createSelector(
+  [selectFilteredEvents],
+  (events) =>
+    events.reduce((previous, event) => {
+      if (event.type !== EventTypes.Port) return [...previous, event]
+
+      const isPortExit = event.portVisitSubEvent === PortVisitSubEvent.Exit
+      const isPortEntry = event.portVisitSubEvent === PortVisitSubEvent.Entry
+      const id = event.id.split('-').shift()
+      const portEntry =
```

Comment:
in the event object you already have the start and end, you dont need the finds

Reply:
In this case the start property of the exit port visit was already overwritten (https://github.com/GlobalFishingWatch/frontend/blob/feature/VV-426/apps/vessel-history/features/resources/resources.slice.ts#L54) so at least the port entry will have to stay.